crates/storage/src/test_fixtures.rs

```diff
@@ -30,13 +30,17 @@ pub(crate) use hash;
 /// Initializers for storage test fixtures.
 pub mod init {
     use super::*;
-    use crate::{StarknetBlocksTable, StarknetStateUpdatesTable};
+    use crate::{
+        state_update::insert_canonical_state_diff, types::CompressedCasmClass,
+        CanonicalBlocksTable, CasmClassTable, ContractCodeTable, StarknetBlocksTable,
+    };
     use pathfinder_common::{ClassCommitment, StorageCommitment};
 
     /// Inserts `n` state updates, referring to blocks with numbers `(0..n)` and hashes `("0x0".."0xn")` respectively.
     pub fn with_n_state_updates(tx: &Transaction<'_>, n: u8) -> Vec<StateUpdate> {
```

Comment:
Maybe we could just throw all of this fixture setup away and use form of "guided" `fake` usage. Ie. specify number and maybe order of entities to be generated and the fixture would also provide methods to easily access the generated content but bypassing the DB (so as to avoid undetected bugs in "roundtrip" db access code).

So creating a fixture would look like this:
- semi-random generation (some params would be provided by the caller, others would rely on `fake`),
- caching in separate structures (lut hashmaps, whatever works),
- then insertion into the db,
- accessor methods used for asserting would look into the cache and ignore the DB which would be fed into UUTs.

Reply:
Yeah I think we can do a better job of creating good data, with a nice builder-like API.

In theory it should be as simple as creating a block header + state update and following the sync process. And of course keep the original data around for the tests to access to get the required output data.